bp_me/src/include/v/bp_mem_wormhole.vh

```diff
@@ -11,58 +11,96 @@
 `include "bsg_noc_links.vh"
 `include "bsg_wormhole_router.vh"
 
-`define declare_bp_mem_wormhole_packet_s(flit_width_mp, cord_width_mp, len_width_mp, cid_width_mp, msg_width_mp, data_width_mp, struct_name_mp) \
+`define declare_bp_mem_wormhole_packet_s(flit_width_mp, cord_width_mp, len_width_mp, cid_width_mp, msg_hdr_name_mp, data_width_mp) \
   typedef struct packed                 \
   {                                     \
-    logic [data_width_mp-1:0] data;     \
-    logic [`bp_mem_wormhole_packet_pad_width(flit_width_mp, cord_width_mp, len_width_mp, cid_width_mp, msg_width_mp)-1:0] \
-                              pad;      \
-    logic [msg_width_mp-1:0]  msg;      \
     logic [cid_width_mp-1:0]  src_cid;  \
     logic [cord_width_mp-1:0] src_cord; \
     logic [cid_width_mp-1:0]  cid;      \
     logic [len_width_mp-1:0]  len;      \
     logic [cord_width_mp-1:0] cord;     \
-  }  struct_name_mp
+  }  bp_mem_router_header_s;            \

```

Comment:
If I understand correctly, the bp_mem_router_header_s is basically a superset of bsg_wormhole_router_header_s as defined in basejump_stl/bsg_noc/bsg_wormhole_router.vh, which has only cord and len fields. We add the others for our specific needs?

Reply:
Yup exactly. The wormhole router itself only considers cord/len.  On top of it, there are other fields like CID that form the "higher level protocol", usually related either to more detailed routing information, or return path formation.

For the mem network, this includes the source cord/cid so that we can return the message properly. In the coh network, the source LCE id is already embedded in the msg_hdr, which is enough to reconstruct the return path. So we just need the cid